### ISSUE TITLE
Uploaded SVGs with <?xml /> tag fail to load on page

### ISSUE THREAD

**Issue by marpstar** [payloadcms/payload#7624]
### Payload Version

3.0.0-beta.73

### Node Version

22.6.0

### Next.js Version

15.0.0-rc.0

### Describe the Bug

Recently upgraded my project to 3.0 beta. Noticed that SVG images being served by Payload are not loading correctly on my page. 

If I hit the URL directly in the address bar, it works fine. Seems to be an issue only when using image tags?

Payload returns 200, Safari will actually let you inspect the DOM of the returned SVG, but it won't show an image preview.

Chrome says that no data was returned but the `Content-Length` response greater reflects the size of the expected SVG. 

[I noticed a comment on Discord](https://discord.com/channels/967097582721572934/1215659716538273832/1240162507193057350) that mentions similar SVG problems. This person found that removing `<?xml version="1.0" encoding="utf-8"?>` from the SVGs fixed the issue.  My results are consistent -- removing that line results in the SVG loading without issue. 

### Reproduction Steps

1. Upload an SVG containing the line `<?xml version="1.0" encoding="utf-8"?>` at the top to a Payload collection. 
2. Use an `<img />` tag with the `src` set to the file URL. 
3. Notice that the image fails to load on the page. 
4. Remove `<?xml version="1.0" encoding="utf-8"?>` from the SVG file
5. SVG file loads without issue

**Comment by lorenzrabs** [comment]
I'm having the same issues. Even with the Image Component from next i'm not able to display SVGs. Removal of "<?xml version="1.0" encoding="utf-8"?>" fixes the issue for me aswell.

**Comment by theo-lubert** [comment]
This is due to the package `file-type` not supporting SVGs

(cf. https://www.npmjs.com/package/file-type)
> This package is for detecting binary-based file formats, not text-based formats like .txt, .csv, .svg, etc.

It might not be an easy fix, but it's worst mentionning that `@payloadcms/storage-s3` does not have that issue.

Most production apps won't use on disk storage anyway. And you can very easily have a [Minio](https://min.io/) running locally if can't use an actual S3 bucket.

**Comment by danielkoller** [comment]
I also ran into this. 

Using a plugin "fixed" it for me. 

Thanks, @theo-lubert.

**Comment by Twansparant** [comment]
Yes same issue here, accessing the image url in the browser works fine, but curling it returns `content-type: application/xml` instead. @linobino1 fix works for my development environment, I'm using cloud storage only in production.

**Comment by blueset** [comment]
I saw that the upload logic has a specific condition to handle this, 

https://github.com/payloadcms/payload/blob/8ace0cab339815627b48f43ee952684f04889ca6/packages/payload/src/uploads/generateFileData.ts#L206C1-L209C6

Perhaps we can also add one in the download logic.

https://github.com/payloadcms/payload/blob/8ace0cab339815627b48f43ee952684f04889ca6/packages/payload/src/collections/endpoints/getFile.ts#L58C1-L61C51

**Comment by manudicri** [comment]
Hi, any news?

**Comment by akaJuliaan** [comment]
+1

**Comment by manudicri** [comment]
Removing the xml tag fixes it for me too..
I guess a custom middleware that removes the tag would be perfect, but I'm new with payload so I don't know if it's possible
